core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -412,28 +497,15 @@ void unpublish() {
         pendingOps.decrementAndGet();
     }
 
-    /**
-     * allocate value handle
-     *
-     * @return if chunk is full return -1, otherwise return new handle index
-     */
-    int allocateHandle() {
-        int hi = handleIndex.getAndIncrement();
-        if (hi + 1 > handles.length) {
-            return -1;
-        }
-        handles[hi] = new Handle();
-        return hi;
-    }
-
     int allocateEntryAndKey(K key) {
         int ei = entryIndex.getAndAdd(FIELDS);
         if (ei + FIELDS > entries.length) {
             return -1;
         }
 
         // key and value must be set before linking to the list so it will make sense when reached before put is done
-        setEntryField(ei, OFFSET_HANDLE_INDEX, -1); // set value index to -1, value is init to null
+        setEntryField(ei, OFFSET.VALUE_POSITION, 0); // set value index to -1, value is init to null
```

Comment:
Why setting separately? Because the entry is not linked and still private? If so please comment

Reply:
Again, didn't come even close to finish the review. Got to linkEntry() of the Chunk. Will continue. This is mostly comment for myself :)

Reply:
This can also be done using unsafe. Will be fixed!